app/uk/gov/hmrc/cgtpropertydisposals/service/TaxEnrolmentService.scala

```diff
@@ -72,4 +78,22 @@ class TaxEnrolmentServiceImpl @Inject()(
       case other => Left(Error(s"Received error response from tax enrolment service with http status: $other"))
     }
 
+  override def hasCgtEnrolment(
+    ggCredId: String
+  )(implicit hc: HeaderCarrier): EitherT[Future, Error, Option[TaxEnrolmentRequest]] =
+    for {
+      maybeEnrolmentRequest <- taxEnrolmentRepository
+                                .get(ggCredId)
+                                .leftMap(error => Error(s"Could not check existence of enrolment in mongo: $error"))
+      taxEnrolmentRequest <- EitherT
+                              .fromOption[Future](maybeEnrolmentRequest, Error("No enrolment request found in mongo"))
+      httpResponse <- makeTaxEnrolmentCall(taxEnrolmentRequest)
+      _ <- EitherT
+            .fromEither[Future](handleTaxEnrolmentResponse(httpResponse))
+            .leftMap(error => Error(s"Could not allocate enrolment: $error"))
+      _ <- taxEnrolmentRepository
+            .delete(ggCredId)
+            .leftMap(error => Error(s"Could not delete enrolment request in mongo: $error"))
```

Comment:
should we surface an error to the user if we can't delete an entry in mongo? It won't break anything - it'll just be some wasted memory in mongo

Reply:
That's my question I posted above. I have asked the GGW team about whether the ES8 API is idempotent. If it is not then this will cause problems and we will have to fix the situation where the enrolment has occurred and we didn't manage to do the delete.

Reply:
Removed the leftMap - we don't need want to surface this to user land.